pkg/scheduler/framework/plugins/clustereviction/cluster_eviction.go

```diff
@@ -0,0 +1,42 @@
+package clustereviction
+
+import (
+	"context"
+
+	"k8s.io/klog/v2"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/scheduler/framework"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+)
+
+const (
+	// Name is the name of the plugin used in the plugin registry and configurations.
+	Name = "ClusterEviction"
+)
+
+// ClusterEviction is a plugin that checks if the target cluster is in the GracefulEvictionTasks which means it has been evicted.
+type ClusterEviction struct{}
+
+var _ framework.FilterPlugin = &ClusterEviction{}
+
+// New instantiates the APIEnablement plugin.
+func New() (framework.Plugin, error) {
+	return &ClusterEviction{}, nil
+}
+
+// Name returns the plugin name.
+func (p *ClusterEviction) Name() string {
+	return Name
+}
+
+// Filter checks if the target cluster is in the GracefulEvictionTasks which means it has been evicted.
+func (p *ClusterEviction) Filter(_ context.Context, bindingSpec *workv1alpha2.ResourceBindingSpec, _ *workv1alpha2.ResourceBindingStatus, cluster *clusterv1alpha1.Cluster) *framework.Result {
```

Comment:
Does this plugin only take effect when the `Application Failover` feature is enabled?

Reply:
Not only application failover, but also the current cluster failover. See comments: https://github.com/karmada-io/karmada/pull/3456#issuecomment-1522660388.

Reply:
Does this conflict with the `TaintToleration` plugin?

Reply:
I don't think there will be any conflict. The plugins will execute sequentially, ensuring that the final result does not include the cluster in eviction tasks, as intended.

Reply:
ok.